cli/gumball-machine/README.md

```diff
@@ -0,0 +1,231 @@
+
+Example `update-header-metadata.json`:
+
+```json
+{
+  "args": {
+    "urlBase": "https://arweave.net",
+    "nameBase": "GUMBALL",
+    "symbol": "GUMBALL",
+    "sellerFeeBasisPoints": 100,
+    "isMutable": true,
+    "retainAuthority": true,
+    "encodeMethod": 1,
+    "price": 10,
+    "goLiveDate": 1234.0,
+    "botWallet": "CwKozhKgAEkxHhJ6AwBn6faFstKWmoFB4JYz18RyxExz",
+    "authority": "Es1EV724ihL7mYzjEiUV1QgcNdwoTpGByLiWSwKFRsAb",
+    "maxMintSize": 10
+  },
+  "optionals": {}
+}
+```
+
+This command does not currently support any optionals, the optionals field can be omitted entirely.
+
+### destroy
+
+Corresponds to `destroy`. Destroys a gumball machine, reclaim rent.
+
+Example usage: destroy a `GumballMachine` on localnet.
+
+`ts-node gumball-machine-cli.ts destroy -u "http://127.0.0.1:8899" -a "/Users/samorend/Documents/candyland/cli/gumball-machine/creator-keypair.json" -g 4SLVDrLyadq6rDb3sKiDR2nX99BiGvXME51Z8B3sPCrE`
+
+Note: destroy does not have any arguments, so -j is not required (or supported) for this command.
+
+### dispense-nft-sol
+
+Corresponds to `dispense_nft_sol` on-chain. Dispenses a certain number of NFTs in exchange for payment in SOL.
+
+Example usage: purchase an NFT with SOL from a `GumballMachine` on localnet.
+
+`ts-node gumball-machine-cli.ts dispense-nft-sol -u "http://127.0.0.1:8899" -g kxTL1hAenEKiZ7vCbRTtH2VBwWDmSntThS6x5UZsPAQ -j "/Users/samorend/Documents/candyland/cli/gumball-machine/example-input-json/dispense-nft-sol.json" -m DT9ejTkku2XrLJoDeNgTY3m9835paesjxqbN2nCzPqgV -r CwKozhKgAEkxHhJ6AwBn6faFstKWmoFB4JYz18RyxExz`
+
+Note: depending on the price of the NFT the receiver might receive such a small payout that it would not be rent exempt and the payout fails. If this occurs, simply airdrop some funds to the receiver pubkey in advance. Also observe that -p is omitted here, because it defaults to the Solana default keypair, which is the account that will receive the compressed NFT.
+
+Example `dispense-nft-sol.json`:
+
+```json
+{
+  "args": {
+    "numItems": 1
+  },
+  "optionals": {}
+}
+```
+
+This command does not currently support any optionals, the optionals field can be omitted entirely.
+
+### dispense-nft-token
+
+Corresponds to `dispense_nft_token` on-chain. Dispenses a certain number of NFTs in exchange for payment in some SPL Token.
+
+Example usage: purchase an NFT with Tokens from a `GumballMachine` on localnet.
+
+`ts-node gumball-machine-cli.ts dispense-nft-token -u "http://127.0.0.1:8899" -g GvpNy2Pw364CYhXk9ws3DMNWzTRe12LUyYP7MpJx7wkx -j "/Users/samorend/Documents/candyland/cli/gumball-machine/example-input-json/dispense-nft-token.json" -m 6rb7AQgVk7BLjoTnD4Hmjo78if2fy7fTRnB6TGkaMxS6 -r H8DWGKyCSaKtgh1GjyqexVcMe8FPyKQoBhKH79QnhDWx -t DKdyWVKLy2XPbSHMXaGBHkFAg59x7V4dRwEvjUrC57me`
```

Comment:
There are a lot of command line args here. As user it is a lot of work to validate the keys... I think it's fine for now, maybe area of improvement for v1

Also, can you remove your fullpath from the JSON arg? Just do write relative path from the `cli/` directory

Reply:
agree. I went back and forth with having the user specify the accounts as options vs. adding that to the JSON. I re-wrote this part of the README with some discussion about it: https://github.com/jarry-xiao/candyland/tree/sorend/gumball-machine-cli/cli/gumball-machine#key-design-choice

But if you find that there's too many options for it to be practical as you're using it then happy to throw the accounts into the JSON as well.